### DISCUSSION TITLE
Example of creating a custom module with custom navigation

### DISCUSSION THREAD

**Discussion by Ali-Dalal** [directus/directus#13932] [Feature Requests]
Hi,
I am trying to create a custom module in Directus with custom navigation. so the navigation bar of the module will also contain some links to other vue components.

for example:
```js
import ModuleComponent from './module.vue';
import AnotherModuleComponent from './another.vue';

export default {
  id: 'custom',
  name: 'Custom',
  icon: 'box',
  routes: [
    {
      path: '/custom',
      component: ModuleComponent,
    },
    {
      path: '/custom2', // I need this to be another component that is reachable via the navigation
      component: AnotherModuleComponent,
    },
  ],
};
```
<img width="703" alt="image" src="https://user-images.githubusercontent.com/12471526/174059797-ae9f5a2c-ae39-4ecb-9c0c-ee77647c41f5.png">
as we can see in the previous image, the module is added but I don't know how to add items into the navigation and define a vue component for each item

Thanks

**Comment by rijkvanzanten** [comment]
The `<private-view>` component relies on Slots to allow you to render Vue components in various sections of the page chrome. In this case, the `#navigation` slot:

https://github.com/directus/directus/blob/c82cc47d3633cef0942e803b50989cb0ea7ee375/app/src/views/private/private-view.vue#L17

```html
<private-view>
  <template #navigation>
    <p>Hi from the nav!</p>
  </template>
</private-view>
```

**Reply by benhaynes** [reply]
CC @erondpowell
